### PR TITLE
[Carousel] Pass index / props down to render items

### PR DESCRIPTION
**Pull request by damassi**
Updates the carousel to pass back props / index so that if needed, we can pass that data along to tracking (or whatever).

### CODE REVIEW

File: src/Components/v2/Carousel.tsx [hunk 1]
```diff
@@ -230,17 +234,21 @@ export class BaseCarousel extends React.Component<
...
     if (!showArrows) {
       return null
     }
 
     const leftPosition =
-      this.props.arrowPosition !== null &&
-      this.props.arrowPosition !== undefined
-        ? this.props.arrowPosition
-        : -space(4)
+      arrowPosition != null ? this.props.arrowPosition : -space(4)
```
Comment (jiahaodavid) [inline src/Components/v2/Carousel.tsx]: I am curious about why did you make this change here? The reason I ask this is the undefined checking seems to make sense. 
Also, you already did the destructing above. When the condiction was true, is there any differences between this.props.arrowPosition and arrowPosition?

Comment (damassi) [reply src/Components/v2/Carousel.tsx]: So, the change with `!== undefined` and `!== `null` to `!=` is a JS shortcut for checking both undefined _and_ null at the same time. Its a weird hack :) 

Re the last `this.props.arrowPosition`, oops! I need to update that spot too. Nice catch 👍

**Comment by xtina-starr** [comment]
Thanks for adding this @damassi!! 🙏

**Comment by artsyit** [comment]
:rocket: PR was released in v16.24.12 :rocket:

**Pull request merged**